CloudFormation/cf_template.yml

```diff
@@ -76,4 +76,37 @@ Resources:
                   - 's3:PutObject'
                 Resource:
                   - 'arn:aws:s3:::my-source-bucket'
-                  - 'arn:aws:s3:::my-destination-bucket/*'
\ No newline at end of file
+                  - 'arn:aws:s3:::my-destination-bucket/*'
+                
+  MyGlueJobRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      RoleName: mz-glue-job-role
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: glue.amazonaws.com
+            Action: 'sts:AssumeRole'
+      Policies:
+        - PolicyName: my-glue-job-policy
+          PolicyDocument:
+            Version: '2012-10-17'
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 's3:GetObject'
+                  - 's3:PutObject'
+                Resource:
+                  - 'arn:aws:s3:::datalake-raw-s3-mz/*'
```

Comment:
two issues here. you are hardcoding the name of the bucket here. if you change the name of the bucket above, you will have to change it here manually, introducing chances for mistakes. use intrinsic functions to construct the arn or retrieve it as I show you in the previous comment.

Also, this role gives glue permission to get and put objects ONLY in folders under these buckets but not directly under the root of this bucket. meaning, your glue job wont be able to put an object at the root of these buckets. it can put objects under folders that are already provisioned inside this bucket. you need to change this like this:

```
- !Sub "arn:aws:s3:::${your bucket resource id}"
- !Sub "arn:aws:s3:::${your bucket resource id}/*"
```